src/Symfony/Component/Serializer/Mapping/Factory/ClassMetadataFactory.php

```diff
@@ -115,4 +115,9 @@ private function getClass($value)
 
         return ltrim(is_object($value) ? get_class($value) : $value, '\\');
     }
+
+    protected function createMetadataClass($class)
```

Comment:
I don't know if it's useful. But if you think it is we could add a new protected method.